api/src/main/java/org/openmrs/module/locationbasedaccess/PersonSearchAdviser.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.locationbasedaccess;
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Person;
+import org.openmrs.PersonAttributeType;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.context.Daemon;
+import org.springframework.aop.Advisor;
+import org.springframework.aop.support.StaticMethodMatcherPointcutAdvisor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class PersonSearchAdviser extends StaticMethodMatcherPointcutAdvisor implements Advisor {
+
+    private static final Log log = LogFactory.getLog(PersonSearchAdviser.class);
+
+    @Override
+    public boolean matches(Method method, Class targetClass) {
+        if (method.getName().equals("getPeople")) {
+            return true;
+        }
+        else if (method.getName().equals("getPerson")) {
+            return true;
+        }
+        else if (method.getName().equals("getPersonByUuid")) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Advice getAdvice() {
+        return new PersonSearchAdvise();
+    }
+
+    private class PersonSearchAdvise implements MethodInterceptor {
+        public Object invoke(MethodInvocation invocation) throws Throwable {
+            if (Context.getAuthenticatedUser() == null) {
+                return null;
+            }
+            Object object = invocation.proceed();
+            if (Daemon.isDaemonUser(Context.getAuthenticatedUser()) || Context.getAuthenticatedUser().isSuperUser()) {
+                return object;
+            }
+            Integer sessionLocationId = Context.getUserContext().getLocationId();
```

Comment:
This looks like a duplicate of what is in PatientSearchAdvice. Can't we refactor it for reuse?

Reply:
Have you mentioned this line or the lines belong to this one?

Reply:
Everything in this method

Reply:
I have analyzed this @dkayiwa  

We can create a common method which can work for persons. So then we can use that to work for Patients. But we need to loop twice the patient list as following,
1. To get the PersonList from the Patient List (common methods needs personList)
2. The common method also loops the personList to check the locations. 

So it may double the time of looping. Do you agree you go for this?

Reply:
Commit and i take a look.

Reply:
I have added a different approach since I need to do the looping for three times earlier (Before calling commongPersonMtd, During the commonPersonMts, after returning from commonPersonMtd)

In the new approach, I have added a new class "AccessLocation" which can carry the needed data for the access. So I can able to reduce the amount of duplicate code in the classes.

Reply:
Reverted to previous step @dkayiwa